app/src/main/java/org/mozilla/fenix/library/history/HistoryViewModel.kt

```diff
@@ -4,34 +4,24 @@
 
 package org.mozilla.fenix.library.history
 
-import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import androidx.paging.LivePagedListBuilder
-import androidx.paging.PagedList
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import kotlinx.coroutines.flow.Flow
 import org.mozilla.fenix.components.history.PagedHistoryProvider
 
 class HistoryViewModel(historyProvider: PagedHistoryProvider) : ViewModel() {
-    var history: LiveData<PagedList<History>>
     var userHasHistory = MutableLiveData(true)
-    private val datasource: LiveData<HistoryDataSource>
-
-    init {
-        val historyDataSourceFactory = HistoryDataSourceFactory(historyProvider)
-        datasource = historyDataSourceFactory.datasource
-
-        history = LivePagedListBuilder(historyDataSourceFactory, PAGE_SIZE)
-            .setBoundaryCallback(object : PagedList.BoundaryCallback<History>() {
-                override fun onZeroItemsLoaded() {
-                    userHasHistory.value = false
-                }
-            })
-            .build()
-    }
-
-    fun invalidate() {
-        datasource.value?.invalidate()
-    }
+    var history: Flow<PagingData<History>> = Pager(
+        PagingConfig(PAGE_SIZE),
+        null
+    ) {
+        HistoryDataSource(
+            historyProvider = historyProvider
+        )
+    }.flow
```

Comment:
Do we need `.cachedIn(viewModelScope)` here? https://developer.android.com/topic/libraries/architecture/paging/v3-paged-data#pagingdata-stream

Reply:
This is necessarily a suggestion to add it, but if it was decided that we don't need to cache the loaded data, we should try to understand what the implication of that is. The docs simply states `The cachedIn() operator makes the data stream shareable and caches the loaded data with the provided CoroutineScope`.

Reply:
It looks like we do not really, it should help survive the orientation state, but it doesn't affect the fenix app because of this:
https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/AndroidManifest.xml#82
Now that I think of it, there is a little use for having a `ViewModel`, there is no needed to support the orientation change. It will work anyway.